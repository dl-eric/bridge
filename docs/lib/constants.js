export const COMPONENTS_URL = '/components';
export const DESIGNS_URL = '/designs';
