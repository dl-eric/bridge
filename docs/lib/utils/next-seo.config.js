export default {
  title: 'Hack4Impact UIUC Design System',
  description: 'Hack4Impact UIUC Design System and React Component Library',
  canonical: 'https://uiuc.hack4impact.org/design',
  openGraph: {
    type: 'website',
    locale: 'en_US',
    url: 'https://uiuc.hack4impact.org/design',
    site_name: 'Hack4Impact UIUC Design System',
  },
  twitter: {
    handle: '@handle',
    site: '@site',
    cardType: 'summary_large_image',
  },
};
