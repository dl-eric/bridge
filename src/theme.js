const colors = {
  black: '#000000',
  white: '#ffffff',
  primary: '#155DA1',

};

const theme = {
  // General
  colors,
};

export default theme;
export { colors };
