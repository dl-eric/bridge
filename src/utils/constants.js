import * as styledSystem from 'styled-system';
import systemPropTypes from '@styled-system/prop-types';

const { compose } = styledSystem;

// common style props to be passed into components
export const COMMON = compose(styledSystem.space, styledSystem.color, styledSystem.layout);
COMMON.propTypes = {
  ...systemPropTypes.space,
  ...systemPropTypes.color,
  ...systemPropTypes.layout,
};
