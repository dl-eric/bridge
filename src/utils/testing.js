import renderer from 'react-test-renderer';

/**
 * Render the component (a React.createElement() or JSX expression)
 * into its intermediate object representation with 'type',
 * 'props', and 'children' keys
 *
 * The returned object can be matched with expect().toEqual(), e.g.
 *
 * ```js
 * expect(render(<Foo />)).toEqual(render(<div foo='bar' />))
 * ```
 */
export function renderJSON(component) {
  return renderer.create(component).toJSON();
}
