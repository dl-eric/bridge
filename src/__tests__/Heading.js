/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { axe, toHaveNoViolations } from 'jest-axe';

import { Heading } from '..';
import theme from '../theme';
import { COMMON, TYPOGRAPHY } from '../utils/constants';
import { renderJSON } from '../utils/testing';
import 'babel-polyfill'; // axe violations required babel-polyfill

expect.extend(toHaveNoViolations);

describe('Heading', () => {
  it('should have no axe violations', async () => {
    const { container } = render(<Heading>Title</Heading>);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    cleanup();
  });

  it('implements system props', () => {
    expect(Heading).toImplementSystemProps(TYPOGRAPHY);
    expect(Heading).toImplementSystemProps(COMMON);
  });

  it('renders a <h2> by default', () => {
    expect(renderJSON(<Heading />).type).toEqual('h2');
  });

  it('respects fontWeight', () => {
    const { container } = render(<Heading fontWeight="bold" />);
    expect(container.firstChild).toHaveStyle(`font-weight: ${theme.fontWeights.bold}px`);

    const { container: c } = render(<Heading fontWeight="regular" />);
    expect(c.firstChild).toHaveStyle(`font-weight: ${theme.fontWeights.regular}px`);
  });

  it('respects the "fontStyle" prop', () => {
    const { container } = render(<Heading fontStyle="italic" />);
    expect(container.firstChild).toHaveStyle('font-style: italic');

    const { container: c } = render(<Heading fontStyle="normal" />);
    expect(c.firstChild).toHaveStyle('font-style: normal');
  });

  it('respects lineHeight', () => {
    const { container } = render(<Heading lineHeight="80px" />);
    expect(container.firstChild).toHaveStyle('line-height: 80px}');
  });

  it('respects fontFamily="main"', () => {
    // styled-components removes the whitespace between font-family values
    const sec = theme.fonts.main.replace(/, /g, ',');
    const { container } = render(<Heading fontFamily="main" />);
    expect(container.firstChild).toHaveStyle(`font-family: ${sec}`);
  });

  it('respects other values for fontSize', () => {
    const { container } = render(<Heading fontSize="2em" />);
    expect(container.firstChild).toHaveStyle('font-size: 2em');

    const { container: c } = render(<Heading fontSize="100px" />);
    expect(c.firstChild).toHaveStyle('font-size: 100px');
  });
});

describe('Heading types', () => {
  it('defaults to type="h2"', () => {
    expect(renderJSON(<Heading>Hello</Heading>)).toMatchSnapshot();
    const { container } = render(<Heading>Hello</Heading>);
    expect(container.firstChild).toHaveStyle(`font-family: ${theme.typography.h2.fontFamily}`);
    expect(container.firstChild).toHaveStyle(`font-weight: ${theme.typography.h2.fontWeight}px`);
    expect(container.firstChild).toHaveStyle(`font-size: ${theme.typography.h2.fontSize}`);
    expect(container.firstChild).toHaveStyle(`letter-spacing: ${theme.typography.h2.letterSpacing}`);
    expect(container.firstChild).toHaveStyle(`line-height: ${theme.typography.h2.lineHeight}`);
  });

  it('renders "h1" properly', () => {
    expect(renderJSON(<Heading type="h1">Hello</Heading>)).toMatchSnapshot();
  });

  it('renders "h3" properly', () => {
    expect(renderJSON(<Heading type="h3">Hello</Heading>)).toMatchSnapshot();
  });

  it('renders "h4" properly', () => {
    expect(renderJSON(<Heading type="h4">Hello</Heading>)).toMatchSnapshot();
  });
});
