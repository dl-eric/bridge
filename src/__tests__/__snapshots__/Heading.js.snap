// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading types defaults to type="h2" 1`] = `
.c0 {
  color: #0A162A;
  font-family: Chivo;
  font-weight: 400px;
  font-size: 44px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 64px;
}

<h2
  className="c0"
  type="h2"
>
  Hello
</h2>
`;

exports[`Heading types renders "h1" properly 1`] = `
.c0 {
  color: #0A162A;
  font-family: Chivo;
  font-weight: 400px;
  font-size: 60px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  line-height: 88px;
}

<h1
  className="c0"
  type="h1"
>
  Hello
</h1>
`;

exports[`Heading types renders "h3" properly 1`] = `
.c0 {
  color: #0A162A;
  font-family: Chivo;
  font-weight: 400px;
  font-size: 32px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 48px;
}

<h3
  className="c0"
  type="h3"
>
  Hello
</h3>
`;

exports[`Heading types renders "h4" properly 1`] = `
.c0 {
  color: #0A162A;
  font-family: Chivo;
  font-weight: 400px;
  font-size: 24px;
  -webkit-letter-spacing: -0.15px;
  -moz-letter-spacing: -0.15px;
  -ms-letter-spacing: -0.15px;
  letter-spacing: -0.15px;
  line-height: 36px;
}

<h4
  className="c0"
  type="h4"
>
  Hello
</h4>
`;
